rtil/src/threads/ue/rebo/rebo_init.rs

```diff
@@ -404,8 +423,22 @@ fn list_recordings() -> Vec<String> {
         }).collect()
 }
 #[rebo::function("Tas::save_recording")]
-fn save_recording(filename: String, recording: Vec<RecordFrame>) {
-    let filename = sanitize_filename::sanitize(filename);
+fn save_recording(filename: String, frames: Vec<RecordFrame>, recording_start_timestamp: u64, recording_end_timestamp: u64) {
+    let recording = Recording {
+        version: 1,
+        author: AMyCharacter::get_player().get_player_name(),
+        steam_id: AMyCharacter::get_player().get_steamid(),
+        filename: filename.clone(),
+        frame_count: frames.len() as i64,
+        recording_start_timestamp,
+        recording_end_timestamp,
+        recording_save_timestamp: SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_millis() as u64,
+        base_speed: AMyCharacter::get_base_speed(),
+        max_walk_speed: AMyCharacter::get_max_walk_speed(),
+        max_bonus_speed: AMyCharacter::get_max_bonus_speed(),
+        frames,
+    };
+    let filename = sanitize_filename::sanitize(filename.clone());
     let path = recording_path().join(filename);
     let file = File::create(path).unwrap();
     serde_json::to_writer_pretty(file, &recording).unwrap();
```

Comment:
Considering we're changing the recording-file format in a backwards-incompatible way, we should add zstd (or some other) compression at the same time. That'll reduce the filesize by a significant factor.
However, it'll also require storing the recording file version in a different way to possibly change the compression at a later point.

Probably fine to change it again at a future point and introduce compression.